include/sdsl/divsufsort.hpp

```diff
@@ -0,0 +1,1973 @@
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef INCLUDED_SDSL_DIVSUFSORT
+#define INCLUDED_SDSL_DIVSUFSORT
+
+namespace sdsl {
+
+#define PROJECT_VERSION_FULL "2.0.1-15-g22e6b23"
+
+#include <assert.h>
+#include <stdio.h>
+#include <inttypes.h> // alternative: #include <stdint.h>
+
+#ifdef _OPENMP
+# include <omp.h>
+#endif
+
+// #include <stdlib.h>
+// #include <memory.h>
+// #include <stddef.h>
+// #include <strings.h>
+
+#if !defined(UINT8_MAX)
+# define UINT8_MAX (255)
+#endif
+
+#define ALPHABET_SIZE (256)
+#define BUCKET_A_SIZE (ALPHABET_SIZE)
+#define BUCKET_B_SIZE (ALPHABET_SIZE * ALPHABET_SIZE)
+#define SS_INSERTIONSORT_THRESHOLD (8)
+#define SS_BLOCKSIZE (1024)
+#define SS_MISORT_STACKSIZE (16)
+#define TR_INSERTIONSORT_THRESHOLD (8)
+
+template <typename saidx_t>
+struct libdivsufsort_config;
+
+template <>
+struct libdivsufsort_config<int32_t>
+{
+    static constexpr uint64_t TR_STACKSIZE = 64;
+    static constexpr uint64_t SS_SMERGE_STACKSIZE = 32;
+};
+
+template <>
+struct libdivsufsort_config<int64_t>
+{
+    static constexpr uint64_t TR_STACKSIZE = 96;
+    static constexpr uint64_t SS_SMERGE_STACKSIZE = 64;
+};
+
+#define BUCKET_A(_c0) bucket_A[(_c0)]
+#if ALPHABET_SIZE == 256
+#define BUCKET_B(_c0, _c1) (bucket_B[((_c1) << 8) | (_c0)])
+#define BUCKET_BSTAR(_c0, _c1) (bucket_B[((_c0) << 8) | (_c1)])
+#else
+#define BUCKET_B(_c0, _c1) (bucket_B[(_c1) * ALPHABET_SIZE + (_c0)])
+#define BUCKET_BSTAR(_c0, _c1) (bucket_B[(_c0) * ALPHABET_SIZE + (_c1)])
+#endif
+
+#define STACK_PUSH(_a, _b, _c, _d)\
+ do {\
+   stack[ssize].a = (_a), stack[ssize].b = (_b),\
+   stack[ssize].c = (_c), stack[ssize++].d = (_d);\
+ } while(0)
+#define STACK_PUSH5(_a, _b, _c, _d, _e)\
+ do {\
+   stack[ssize].a = (_a), stack[ssize].b = (_b),\
+   stack[ssize].c = (_c), stack[ssize].d = (_d), stack[ssize++].e = (_e);\
+ } while(0)
+#define STACK_POP(_a, _b, _c, _d)\
+ do {\
+   assert(0 <= ssize);\
+   if(ssize == 0) { return; }\
+   (_a) = stack[--ssize].a, (_b) = stack[ssize].b,\
+   (_c) = stack[ssize].c, (_d) = stack[ssize].d;\
+ } while(0)
+#define STACK_POP5(_a, _b, _c, _d, _e)\
+ do {\
+   assert(0 <= ssize);\
+   if(ssize == 0) { return; }\
+   (_a) = stack[--ssize].a, (_b) = stack[ssize].b,\
+   (_c) = stack[ssize].c, (_d) = stack[ssize].d, (_e) = stack[ssize].e;\
+ } while(0)
+
+static const int32_t lg_table[256]= {
+ -1,0,1,1,2,2,2,2,3,3,3,3,3,3,3,3,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,
+  5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,
+  6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,
+  6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,
+  7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,
+  7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,
+  7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,
+  7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7
+};
+
+#if (SS_BLOCKSIZE == 0) || (SS_INSERTIONSORT_THRESHOLD < SS_BLOCKSIZE)
```

Comment:
Both macros are fixed values in this file so either this condition is always true or always false

Reply:
I'd keep this as well just in case.